### DISCUSSION TITLE
How to update finch without uninstalling on macOS?

### DISCUSSION THREAD

**Discussion by n-guitar** [runfinch/finch#250] [Q&A] [answered]
I'm using Finch v0.2.0 on macOS.
How to update finch without uninstalling on macOS?
I'm planning to update from v0.2.0 to v0.4.0.
If possible, I would like to build a container image that has already been created without deleting it, so I asked the question.

**Additional context**
```bash
$finch --version
Finch version v0.2.0

# M1 Macbook pro
$ sw_vers
ProductName: macOS
ProductVersion: 12.6.3
buildVersion: 21G419
```

**Comment by AnqiPang** [comment]
Hi, I will convert this issue to discussion and give updates there. Thanks!

**Comment by AnqiPang** [comment, marked as answer]
Hi @n-guitar, you could follow the flow below to restore your image after upgrade to v0.4.0 from v0.2.0. You could back up your data before your actually execute the commands below in case encounter unexpected errors.
1. Uninstall v0.2.0 and install v0.4.0.
2. Run the commands below to mount the correct data disk.
```
finch vm init
finch vm stop
export NEW_FINCH_DISK=$(readlink /Applications/Finch/lima/data/_disks/finch/datadisk)
mv ~/.finch/.datadisk $NEW_FINCH_DISK
finch vm start
```
3. Shell into the VM.
```
LIMA_HOME=/Applications/Finch/lima/data /Applications/Finch/lima/bin/limactl shell finch
```
4. Move containerd user data to correct directories.
```
sudo rm -r /mnt/lima-finch/containerd/
sudo mkdir -p /mnt/lima-finch/containerd
sudo mv /mnt/lima-finch/io.* /mnt/lima-finch/tmpmounts/ /mnt/lima-finch/containerd/
```

 5. Logout the VM and run `finch vm stop && finch vm start`. Then verify if your images persist with `finch images`.
 
Thanks for @davidhsingyuchen's help to figure out the solution.
